exchanges/poloniex/poloniex.go

```diff
@@ -3,933 +3,1124 @@ package poloniex
+			return nil, err
+		}
 	}
-	if len(clientOrderIDs) > 0 {
-		values.Set("clientOrderIds", strings.Join(clientOrderIDs, ","))
+	params := url.Values{}
+	if !startTime.IsZero() {
+		params.Set("startTime", strconv.FormatInt(startTime.UnixMilli(), 10))
+	}
+	if !endTime.IsZero() {
+		params.Set("endTime", strconv.FormatInt(endTime.UnixMilli(), 10))
+	}
+	if len(symbols) != 0 {
+		params.Set("symbols", symbols.Join())
+	}
+	if from > 0 {
+		params.Set("from", strconv.FormatUint(from, 10))
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatUint(limit, 10))
+	}
+	if direction != "" {
+		params.Set("direction", direction)
+	}
+	var resp []*TradeHistory
+	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, sGetTradesEPL, http.MethodGet, "/trades", params, nil, &resp)
+}
+
+// GetTradesByOrderID gets trades for an order
+func (e *Exchange) GetTradesByOrderID(ctx context.Context, orderID string) ([]*TradeHistory, error) {
+	if orderID == "" {
+		return nil, order.ErrOrderIDNotSet
 	}
-	var result []CancelOrdersResponse
-	return result, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodDelete,
-		poloniexCancelByIDs,
-		values,
-		&result)
+	var resp []*TradeHistory
+	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, sGetTradeDetailEPL, http.MethodGet, "/orders/"+orderID+"/trades", nil, nil, &resp)
 }
 
 // SendHTTPRequest sends an unauthenticated HTTP request
-func (e *Exchange) SendHTTPRequest(ctx context.Context, ep exchange.URL, path string, result any) error {
+func (e *Exchange) SendHTTPRequest(ctx context.Context, ep exchange.URL, epl request.EndpointLimit, path string, result any) error {
 	endpoint, err := e.API.Endpoints.GetURL(ep)
 	if err != nil {
 		return err
 	}
-
-	item := &request.Item{
-		Method:                 http.MethodGet,
-		Path:                   endpoint + path,
-		Result:                 result,
-		Verbose:                e.Verbose,
-		HTTPDebugging:          e.HTTPDebugging,
-		HTTPRecording:          e.HTTPRecording,
-		HTTPMockDataSliceLimit: e.HTTPMockDataSliceLimit,
+	resp := result
+	if strings.HasPrefix(path, v3Path) {
+		resp = &V3ResponseWrapper{
+			Data: result,
+		}
 	}
-
-	return e.SendPayload(ctx, request.UnAuth, func() (*request.Item, error) {
-		return item, nil
-	}, request.UnauthenticatedRequest)
+	if err := e.SendPayload(ctx, epl, func() (*request.Item, error) {
+		return &request.Item{
+			Method:                 http.MethodGet,
+			Path:                   endpoint + path,
+			Result:                 resp,
+			Verbose:                e.Verbose,
+			HTTPDebugging:          e.HTTPDebugging,
+			HTTPRecording:          e.HTTPRecording,
+			HTTPMockDataSliceLimit: e.HTTPMockDataSliceLimit,
+		}, nil
+	}, request.UnauthenticatedRequest); err != nil {
+		return err
+	}
+	if resp == nil {
+		return common.ErrNoResponse
+	}
+	if errType, ok := result.(interface{ Error() error }); ok {
+		return errType.Error()
+	}
```

Comment:
🚧 Change request:
Same request as before.
* ~Please check `result` not `resp` for `nil`~
* Please check `resp` for the Error, not `result`
* **And actually, I think we can simplify things. The Error functions can do the nil check as appropriately for the type!**
```suggestion
	if errType, ok := resp.(interface{ Error() error }); ok && errType.Error() != nil {
		return errType.Error()
	}
```